src/Analysis/GWASResultsV2/Views/Execution/Execution.stories.jsx

```diff
@@ -1,28 +1,124 @@
   alert(`setCurrentView called with ${input}`);
 };
+const selectedRowData = {
+  name: 'gwas-workflow-787571537',
+  wf_name: 'user created name',
+  uid: '4b125c09-9712-486f-bacd-ec1451aae935',
+  startedAt: '2022-02-15T13:00:00Z',
+  finishedAt: '2022-02-15T14:00:00Z',
+  phase: 'Failed',
+  DateTimeSubmitted: '2022-02-15T13:30:00Z',
+};
+const { name, uid } = selectedRowData;
+
+const MockedFailureJSON = [
+  {
+    name: 'gwas-workflow-7875715375',
+    step_template: 'gwas-workflow',
+    error_message: 'No message provided',
+  },
+  {
+    name: 'gwas-workflow-7875715375.get-pheno-csv',
+    step_template: 'get-pheno-csv',
+    error_message: 'pod deleted',
+  },
+];
+
+const MockTemplateFailure = () => {
+  return (
+    <QueryClientProvider client={mockedQueryClient}>
+      <SharedContext.Provider
+        value={{
+          selectedRowData,
+          setCurrentView,
+        }}
+      >
+        <Execution />
+      </SharedContext.Provider>
+    </QueryClientProvider>
+  );
+};
 
-const MockTemplate = () => {
+const MockTemplateSuccess = () => {
   return (
-    <SharedContext.Provider
-      value={{
-        selectedRowData,
-        setCurrentView,
-      }}
-    >
-      <Execution />
-    </SharedContext.Provider>
+    <QueryClientProvider client={mockedQueryClient}>
+      <SharedContext.Provider
+        value={{
+          selectedRowData: { ...selectedRowData, phase: 'Succeeded' },
+          setCurrentView,
+        }}
+      >
+        <Execution />
+      </SharedContext.Provider>
+    </QueryClientProvider>
   );
 };
 
-export const MockedSuccess = MockTemplate.bind({});
+export const MockedFailure = MockTemplateFailure.bind({});
+MockedFailure.parameters = {
+  msw: {
+    handlers: [
+      rest.get(
+        `http://:argowrapperpath/ga4gh/wes/v2/logs/${name}?uid=${uid}`,
+        (req, res, ctx) => {
+          const { argowrapperpath } = req.params;
+          return res(ctx.delay(100), ctx.json(MockedFailureJSON));
+        }
+      ),
+    ],
+  },
+};
+
+export const MockedSuccess = MockTemplateSuccess.bind({});
+MockedSuccess.parameters = {
+  msw: {
+    handlers: [
+      rest.get(
+        `http://:argowrapperpath/ga4gh/wes/v2/logs/${name}?uid=${uid}`,
+        (req, res, ctx) => {
+          const { argowrapperpath } = req.params;
+          // Successful executions return an empty array
+          return res(ctx.delay(100), ctx.json([]));
+        }
+      ),
+    ],
+  },
+};
+
+export const MockedError = MockTemplateFailure.bind({});
+MockedError.parameters = {
+  msw: {
+    handlers: [
+      rest.get(
+        `http://:argowrapperpath/ga4gh/wes/v2/logs/${name}?uid=${uid}`,
+        (req, res, ctx) => {
+          const { argowrapperpath } = req.params;
+          return res(
+            ctx.delay(100),
+            // Some errroneous responses can return an error object
+            ctx.json({ error: 'Mocked Server error response' })
+          );
+        }
+      ),
+    ],
+  },
+};
```

Comment:
Please also add a scenario where the endpoint itself fails (403)

Reply:
Will do this next sprint, thank you!